controllers/azuremanagedcontrolplane_reconciler.go

```diff
@@ -138,20 +139,25 @@ func (r *azureManagedControlPlaneService) reconcileKubeconfig(ctx context.Contex
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "controllers.azureManagedControlPlaneService.reconcileKubeconfig")
 	defer done()
 
-	kubeConfigData := r.scope.GetKubeConfigData()
```

Comment:
@LochanRn fyi this change conflicts with #4052 so whichever PR doesn't merge first will have to do some rebasing

Reply:
@CecileRobertMichon  okay seems like a big piece, and a great one !!